### DISCUSSION TITLE
Add Incomplete Save Path Option for Torrent Clients

### DISCUSSION THREAD

**Discussion by budgetfermium** [autobrr/autobrr#1647] [Ideas]
**Is your feature request related to a problem? Please describe.**
Under the Actions tab, it is currently only possible to set the save path.

**Describe the solution you'd like**
I would like to be able to set the incomplete save path when adding torrents to a torrent client.

**Comment by zze0s** [comment]
Hi!

Any specific client you're using? This is usually done with settings in the client itself. When autobrr hands over the torrent to the client it's job is done there.

Not all clients even support setting that but Deluge seem to have the option.

For qbit it's recommended to either use the built in settings for it, or use categories with different paths and then use a 3rd party tool like `qbittools`, `qbt-cli`, `qbit_manage` or similar which have functions to change the category after X time or torrent age.

**Comment by budgetfermium** [comment]
I am using qBittorrent. As of Web API 2.8.4, it is possible to set a "download_path" parameter to specify the desired incomplete directory, so it should be feasible to have a simple implementation without relying on third-party tools. I tried to do it myself, but cannot get it to work, especially without updated documentation at [qBittorrent's WebUI API reference page](https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)).

**Comment by zze0s** [comment]
> I am using qBittorrent. As of Web API 2.8.4, it is possible to set a "download_path" parameter to specify the desired incomplete directory, so it should be feasible to have a simple implementation without relying on third-party tools. I tried to do it myself, but cannot get it to work, especially without updated documentation at [qBittorrent's WebUI API reference page](https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)).

Do you have any links to issues or prs for this? I looked briefly but didn't find anything useful. Their API docs are ok but seem to be missing things here and there.

**Comment by budgetfermium** [comment]
qBittorrent got support for setting download paths [in this commit](https://github.com/qbittorrent/qBittorrent/pull/15793/commits/1c0f8b4289b8d298041fe195cdc24f793dce4df5) - see [here](https://github.com/qbittorrent/qBittorrent/pull/15793) for accompanying discussion. A Python implementation is available [here](https://github.com/rmartin16/qbittorrent-api/pull/60) for reference.
